workspace/gh-workflow/src/generate.rs

```diff
@@ -6,16 +6,19 @@ use derive_setters::Setters;
 use crate::error::{Error, Result};
 use crate::Workflow;
 
-#[derive(Setters, Clone)]
+const DEFAULT_WORKFLOW_NAME: &str = "ci.yml";
+
+#[derive(Setters)]
 #[setters(into)]
 pub struct Generate {
     workflow: Workflow,
-    name: String,
+    name: Option<String>,
 }
 
 impl Generate {
-    pub fn new(workflow: Workflow) -> Self {
-        Self { workflow, name: "ci.yml".to_string() }
+    pub fn new(workflow: Workflow, name: Option<String>) -> Self {
```

Comment:
I'm new to Rust - is this a desirable API or would it make more sense to require this parameter? Is the `String` the right datatype?